compiler/src/main/kotlin/io/github/apl_cornell/viaduct/syntax/circuit/Statements.kt

```diff
@@ -0,0 +1,196 @@
+package io.github.apl_cornell.viaduct.syntax.circuit
+
+import io.github.apl_cornell.viaduct.prettyprinting.Document
+import io.github.apl_cornell.viaduct.prettyprinting.bracketed
+import io.github.apl_cornell.viaduct.prettyprinting.concatenated
+import io.github.apl_cornell.viaduct.prettyprinting.joined
+import io.github.apl_cornell.viaduct.prettyprinting.nested
+import io.github.apl_cornell.viaduct.prettyprinting.plus
+import io.github.apl_cornell.viaduct.prettyprinting.times
+import io.github.apl_cornell.viaduct.syntax.Arguments
+import io.github.apl_cornell.viaduct.syntax.SourceLocation
+import io.github.apl_cornell.viaduct.syntax.surface.keyword
+import kotlinx.collections.immutable.PersistentList
+import kotlinx.collections.immutable.toPersistentList
+
+/** A computation with side effects. */
+sealed class StatementNode : Node()
+
+/**
+ * A statement that is _not_ a combination of other statements, and that does not affect
+ * control flow.
+ */
+sealed class SimpleStatementNode : StatementNode()
```

Comment:
There won't be any control flow statements in this IR since we will use CPS. So there is no need for `SimpleStatementNode`. We have `StatementNode`s used in general functions, and `CircuitStatementNode`s used in circuit blocks.